### ISSUE TITLE
i can not add a new record in new generated entity

### ISSUE THREAD

**Issue by ruizhit** [michaelkrone/generator-material-app#25]
Hi i got this issue

I click on create one.
The app shows create modal, but immediately closes the modal
without giving me a chance to add the record

No errors in developer console

![closewindow](https://cloud.githubusercontent.com/assets/7987506/10677009/8079f068-78cf-11e5-8e68-9baf7e2d78df.png)

immediately closes the modal

<img width="1271" alt="screen shot 2015-10-22 at 15 16 59" src="https://cloud.githubusercontent.com/assets/7987506/10677105/0db3e57e-78d0-11e5-98ba-7abebcaf59fb.png">

**Comment by michaelkrone** [comment]
Hi @ruizhit.
Thank you, I will have a look at this at well.

**Comment by ricsard** [comment]
@ruizhit I also have this problem, as I see we faced with the same issues, I didn't find the solution for this yet, but I would appreciate if we find a solution for sg we will share it here :)
@michaelkrone :+1:

**Comment by Bernix01** [comment]
Hi, in my case for some reason the paths are generated with "\" instead of "/" which got scaped causing the dialog to hide... the workaround was going through 

entity.js
main/main.js
list/list.js
list/detail/detail.js
list/edit/edit.js

and change the "\" with "/" after that the dialog didn't dissappear.

**Comment by michaelkrone** [comment]
Hi @Bernix01. Which paths were generated with a "\"? The routes?

**Comment by Bernix01** [comment]
@michaelkrone those for the html templates.

**Comment by tenchugecko** [comment]
I got it working, but the create immediately hides again, after changing the server model

**Comment by tenchugecko** [comment]
Ok got the update thingy:

 at \server\lib\controllers\param.controller.js:114:12
Search for "this" inside the mongodbCrud closures and do the 
var self = this 
to get this into the scope of the closure

'''
update: function (req, res) {
	 
	    var self = this;

		if (req.body._id) {
			delete req.body._id;
		}

		var self = this;
		var bodyData = _.omit(req.body, this.omit);
		var updated = _.merge(req[this.paramName], bodyData);

		updated.save(function (err) {
			if (err) {
				return res.handleError(err);
			}

			req[self.paramName] = updated;
			return res.ok(self.getResponseObject(updated));
		});
	},
'''

**Comment by tenchugecko** [comment]
the other thing about that create window auto hiding

removing the hide from statechanged "fixes" it, so  - what is it i'm not understanding here :)
```javascript
function onEnterScheduleListCreateView($rootScope, $state, $mdDialog) {
		var unregisterListener = $rootScope.$on('$stateChangeStart', onStateChange);

		$mdDialog.show({
			controller: 'ScheduleCreateController',
			controllerAs: 'create',
			templateUrl: 'app/schedules/create/create.html',
			clickOutsideToClose: false
		}).then(transitionTo, transitionTo);

		/**
		 * Function executed when resolving or rejecting the
		 * dialog promise.
		 *
		 * @param {*} answer - The result of the dialog callback
		 * @returns {promise}
		 */
		function transitionTo(answer) {
			return $state.transitionTo('schedule.list');
		}

		/**
		 * Function executed when changing the state.
		 * Closes the create dialog
		 */
		function onStateChange() {
			unregisterListener();
		//	$mdDialog.hide();
		}
	}
`````

**Comment by Bernix01** [comment]
As @tenchugecko suggested

```javascript
         /**
         * Function executed when changing the state.
         * Closes the create dialog
         */
        function onStateChange() {
            unregisterListener();
        //  $mdDialog.hide();
        }
```

fixes the auto hiding problem now all seems to work.

**Comment by tenchugecko** [comment]
so : why is the onStateChanged fired right away on these views

 and not why is it not fired on the Admin/Users view.

**Comment by michaelkrone** [comment]
Should be closed with #38
